README.md

```diff
@@ -1,28 +1,103 @@
 # StableHLO
 
-StableHLO is an operation set that expresses ML computations. It has been
-originally bootstrapped from
-[the MHLO dialect](https://github.com/tensorflow/mlir-hlo#meta-hlo-dialect-mhlo)
-and enhances it with additional functionality, including serialization and
-versioning.
+StableHLO is an operation set for high-level operations (HLO) in machine
+learning (ML) models. Essentially, it's a portability layer between different
+ML frameworks and ML compilers: ML frameworks that produce StableHLO programs
+are compatible with ML compilers that consume StableHLO programs.
 
-StableHLO is a portability layer between ML frameworks and ML compilers.
-We are aiming for adoption by a wide variety of ML frameworks including
-TensorFlow, JAX and PyTorch, as well as ML compilers including XLA and IREE.
+Our goal is to simplify and accelerate ML development by creating more
+interoperability between various ML frameworks (such as TensorFlow, JAX and
+PyTorch) and ML compilers (such as XLA and IREE).
 
-## Development
+StableHLO is based on the MHLO dialect and enhances it with additional
+functionality, including serialization and versioning. This provides
+[backward and forward
+compatibility](https://github.com/openxla/stablehlo/blob/main/rfcs/20220912-compatibility.md)
+guarantees for StableHLO programs and ensures compatibility between frameworks
+and compilers, even as StableHLO continues to evolve.
+
+This repository includes the [StableHLO
+specification](https://github.com/openxla/stablehlo/blob/main/docs/spec.md)
+along with implementations in C++ and Python, which you can use to build
```

Comment:
It is not clear what the input is, using the implementation, to produce a StableHLO programs.

Reply:
I'm not sure I understand. Are you suggesting that we explain here that the ML frameworks are the source for the ML programs and that's where these implementations are used? I think that would be overly specific and, IIRC, @burmako prefers that we not be prescriptive about who/what is actually building the StableHLO programs. But I might have misunderstood that. 

I revised this sentence slightly for clarity, but I don't think it addresses your concern. As is, this is intentionally ambiguous about who defines the actual ML program: This is just an API to build ML programs/graphs, so it can be used to write programs by hand or be implemented in another program (an ML framework) to generate the StableHLO output using some other input. I expect engineers will understand that, but if there's another way you think we can make that more clear, I'm happy to revise this.

Reply:
Ps.
As we don't care to detail if are the Frameworks bridges developers to orchestrate these API or a developer directly to express his own ML program to obtain a StableHLO ML program output I think my previous example is general enough.

Reply:
I think I understand. I've now added that this StableHLO program can be fed to XLA/IREE, but still without specifying how the StableHLO is literally produced. Sound good?